src/Compilers/CSharp/Portable/Symbols/PublicModel/PreprocessingSymbol.cs

```diff
@@ -65,34 +73,49 @@ bool ISymbol.Equals(ISymbol? other, CodeAnalysis.SymbolEqualityComparer equality
 
         Accessibility ISymbol.DeclaredAccessibility => Accessibility.NotApplicable;
 
-        void ISymbol.Accept(SymbolVisitor visitor) => throw new System.NotSupportedException();
+        void ISymbol.Accept(SymbolVisitor visitor)
+        {
+            throw new NotSupportedException();
+        }
 
-        TResult ISymbol.Accept<TResult>(SymbolVisitor<TResult> visitor) => throw new System.NotSupportedException();
+        TResult ISymbol.Accept<TResult>(SymbolVisitor<TResult> visitor)
+        {
+            throw new NotSupportedException();
+        }
 
-        TResult ISymbol.Accept<TArgument, TResult>(SymbolVisitor<TArgument, TResult> visitor, TArgument argument) => throw new System.NotSupportedException();
+        TResult ISymbol.Accept<TArgument, TResult>(SymbolVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            throw new NotSupportedException();
+        }
 
         string? ISymbol.GetDocumentationCommentId() => null;
 
         string? ISymbol.GetDocumentationCommentXml(CultureInfo? preferredCulture, bool expandIncludes, CancellationToken cancellationToken) => null;
 
         string ISymbol.ToDisplayString(SymbolDisplayFormat? format)
         {
-            return SymbolDisplay.ToDisplayString(this, format);
+            return _name;
```

Comment:
this seems weird.  by not calling into SymbolDisplay, you make it so that if someone uses taht public API, they get what now?  nothing for tehse symbols?  (the same applies to the changes below).

Reply:
Previously an exception was being thrown, as these symbols were not properly supported. Cannot recall the exact issue that arose, but this adjustment was necessary to avoid going into the SymbolDisplay path. An alternative approach could work, if necessary.

Reply:
you should just go the symbol-display path.  It doesn't really make sense that you could call `ppSymbol.ToDisplayString()` on the symbol and have it work, but then fail if you called `SymbolDisplay.ToDisplayString(ppSymbol)`.

Note: it's unclear what this is currently needed for at all though.  Where is this used in FAR?

Reply:
As commented below too, this is part of also supporting the LSP side of the implementation, for which a test was failing.

I'm changing the implementation to go the SymbolDisplay path, so as to have a common flow.